### PR TITLE
Changes to Die, Game, Program and Testing classes

### PR DESCRIPTION
**Pull request by scott-bree**
Added Roll class to roll random 1-6 dies. Updated classes to test and run 3 dice rolls and print sum to console.

### CODE REVIEW

**Review by Hyperfied — APPROVED** [review]
- You've met the main requirements
- The code is styled following the Style Guidelines
- The code is easy to read
- Errors are handled with testing
To improve you could use XML comments for the functions to better explain what they do.

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -7,17 +7,17 @@
...
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
-
-        //Property
-
 
-        //Method
+    {
+    // Die value property
+    private int DieValue = 0;
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Die.cs]: DieValue should be written in camelCase.

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -8,14 +8,36 @@ namespace CMP1903_A1_2324
...
+            // Create random object
+            Random random = new Random();
+            
+            // Create 3 dice objects
+            Die die1 = new Die();
+            Die die2 = new Die();
+            Die die3 = new Die();
+            
+            // Roll the dices
+            var roll1 = die1.Roll(random);
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Game.cs:23]: You alternate between explicitly-typed and implicitly-typed variables (var and Die), try sticking to one.

File: CMP1903_A1_2324/Game.cs [hunk 3]
```diff
@@ -8,14 +8,36 @@ namespace CMP1903_A1_2324
...
+            var roll2 = die2.Roll(random);
+            var roll3 = die3.Roll(random);
+            
+            // Write dice outputs to console
+            Console.WriteLine("Roll 1: " + roll1);
+            Console.WriteLine("Roll 2: " + roll2);
+            Console.WriteLine("Roll 3: " + roll3);
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Game.cs:30]: Personally I wouldn't write to the console in a class, I'd only write to the console in a main function.

File: CMP1903_A1_2324/Game.cs [hunk 4]
```diff
@@ -8,14 +8,36 @@ namespace CMP1903_A1_2324
...
+            
+            // Sum all the dice rolls
+            int total = roll1 + roll2 + roll3;
+            return total;
+        }
+        
+        // write the output of RollDice to the console
+        public void ReportTotal(int total)
+        {
+            Console.WriteLine("The total of the three dice rolls is: " + total);
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Game.cs]: This could just be a line in the main class.

File: CMP1903_A1_2324/Program.cs [hunk 5]
```diff
@@ -10,10 +10,14 @@ internal class Program
     {
         static void Main(string[] args)
         {
-            /*
-             * Create a Game object and call its methods.
-             * Create a Testing object to verify the output and operation of the other classes.
-             */
+            // Creates a game object and calls its methods
+            Game game = new Game();
+            var total = game.RollDice();
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Program.cs]: Again swapping between explicitly and implicitly typed.

File: CMP1903_A1_2324/Testing.cs [hunk 6]
```diff
@@ -8,13 +9,22 @@ namespace CMP1903_A1_2324
...
+            
+            // Create game object
+            Game game = new Game();
+            
+            // Die rolls are between 1 and 6 (inclusive)
+            Die die = new Die();
+            Debug.Assert(die.Roll(random) >= 1 && die.Roll(random) <= 6, "Die roll is not between 1 and 6");
+            
+            // The total of the three dice rolls is between 3 and 18 (inclusive)
+            var total = game.RollDice();
```
Comment (Hyperfied) [inline CMP1903_A1_2324/Testing.cs:26]: Another implicitly typed along with explicit.

**Review by Hyperfied — APPROVED** [review]
- You've met the main requirements
- Most of the code follows the Style Guidelines
- The code is very readable
- There is error checking
To improve further you could add XML comments to make what the functions do more clear.

**Review by magoed12 — COMMENTED** [review]
Code meets all requirements, is well structured and uses error checking to prevent crashes. Comments are again well structured, appropriate and do a good job of explaining what is happening at any given point without being too long. The Code also fits well with the C# style sheet. Outputs could be altered so they are all in the same place therefore making them easier to edit and formatted, for example only have Console.Writeline() in one class or the main program.

**Review by Hyperfied — APPROVED** [review]
You've improved on the comments I've made, well done!!!

**Pull request merged**